rotkehlchen/chain/evm/decoding/cowswap/decoder.py

```diff
@@ -328,20 +317,36 @@ def _aggregator_post_decoding(
 
         return decoded_events
 
+    @staticmethod
+    def _match_cowswap_counterparty(context: DecoderContext) -> DecodingOutput:
+        """Sets the CowSwap counterparty for events involving the settlement contract.
+
+        This is needed because CowSwap solvers execute transactions without
```

Comment:
You also need to rephrase this I guess. The problem is simply that if it's a smart account you won't detect it since the settlement contract won't be the to_address of the transaction as you had before?